src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -265,9 +266,25 @@ object RegressionUtils {
 
     lrb.stats(y, nSamples, minAC)
   }
+
+  // constructs SparseVector of hard call genotypes (with missing values mean-imputed) in parallel with other summary statistics
+  // if all genotypes are missing then all elements are NaN
+  def toLinMixedHardCallStats(gs: Iterable[Genotype], mask: Array[Boolean], nSamples: Int): SparseGtVectorAndStats = {
+    val sb = new SparseGtBuilder()
+    val gts = gs.hardCallGenotypeIterator
+
+    var i = 0
+    while (i < mask.length) {
```

Comment:
nSamples is the number of filtered samples in the regression (the length of y).

mask.length is the number of samples in the unfiltered vds.

The kept samples have mask(i) == true